OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Options;
+using OrderService.Domain.Abstractions.Data;
+using OrderService.Domain.Entities;
+using System.Text.Json;
+
+namespace OrderService.Infrastructure.Services;
+
+internal class RedisStorageService : ITemporaryStorageService
+{
+    private readonly IDistributedCache _cache;
+    private readonly HttpContext _httpContext;
+
+    public RedisStorageService(IDistributedCache cache, IHttpContextAccessor httpContextAccessor)
+    {
+        _cache = cache;
+        _httpContext = httpContextAccessor.HttpContext;
+    }
+ 
+    public async Task<Dictionary<int, ProductEntity>> GetCartAsync(CancellationToken cancellationToken = default)
+    {
+        var cartId = _httpContext.User.FindFirst("Id")?.Value ?? _httpContext.Request.Cookies["CartId"];
+
+        if(cartId is null)
+        {
+            cartId = Guid.NewGuid().ToString();
+            _httpContext.Response.Cookies.Append("CartId", cartId, new CookieOptions
+            {
+                Expires = DateTimeOffset.UtcNow.AddDays(30), 
+                HttpOnly = true, 
+                Secure = true, 
+            });
+        }
+
+        var userId = _httpContext.User.FindFirst("Id")?.Value;
```

Comment:
you can do it in the beginning of the method and use userId in finding of cartId